src/components/taskDetails/TaskDates.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import Details from './Details';
+import styles from './task-details.module.scss';
+import { TASK_EXTENSION_REQUEST_URL } from '@/constants/url';
+import convertTimeStamp from '@/helperFunctions/convertTimeStamp';
+
+interface TaskDatesProps {
+    isEditing: boolean;
+    isUserAuthorized: boolean;
+    startedOn: string;
+    endsOn: number;
+    newEndOnDate: string;
+    setNewEndOnDate: (date: string) => void;
+    handleBlurOfEndsOn: () => void;
+    isExtensionRequestPending: boolean;
+    taskId: string;
+}
+
+export const TaskDates: React.FC<TaskDatesProps> = ({
+    isEditing,
+    isUserAuthorized,
+    startedOn,
+    endsOn,
+    newEndOnDate,
+    setNewEndOnDate,
+    handleBlurOfEndsOn,
+    isExtensionRequestPending,
+    taskId,
+}) => {
+    const formattedEndsOn = endsOn ? convertTimeStamp(endsOn) : 'TBD';
+
+    return (
+        <>
+            <div className={styles.inputContainer}>
+                <Details detailType="Started On" value={startedOn} />
+            </div>
+            <div className={styles.inputContainer}>
+                {isExtensionRequestPending && (
```

Comment:
Wouldn't it be better to use ternary operator here?

like this
```js
{isExtensionRequestPending ? (
        <Details
            detailType="Ends On"
            value={formattedEndsOn}
            url={`${TASK_EXTENSION_REQUEST_URL}?&q=${encodeURIComponent(`taskId:${taskId},status:PENDING`)}`}
        />
    ) : (
        <Details detailType="Ends On" value={formattedEndsOn} />
    )} ```

Reply:
The current approach of using separate `if` conditions instead of a ternary operator is deliberate to enhance code readability and maintainability. By separating the conditions, it is easier to understand the logic at a glance, especially when dealing with nested or multiple conditional checks.

This method also simplifies future modifications or extensions, where additional conditions or elements might be added without nesting ternary operators, which can sometimes lead to more complex and less readable code.

Reply:
Yes, I agree to your point but in future we can again refactor no, I am resolving this btw